### DISCUSSION TITLE
Can we seperate each out put parameters?

### DISCUSSION THREAD

**Discussion by ZhixinFan-hub** [epochpic/epoch#640] [General]
Hi
When I was using EPOCH simulations, I realized that the output size for species like `px, py, vx, vy, gamma, id` is huge, maybe tens to hundreds GB. It's hard to download to my desktop and analyze on matlab.
So I wonder is there anyway that can reduce the file size? Can we seperate each out parameters into different files?

**Comment by Status-Mirror** [comment]
Hey @ZhixinFan-hub,

When files become too large, you have a few options. Here are some examples:
- You can have multiple output blocks writing to different files at different frequencies. See the "multiple output blocks" section in the [output block](https://epochpic.github.io/documentation/input_deck/input_deck_output_block.html) documentation.
- Add `+ single` onto your output lines, to force single-precision dumps instead of double precision. This halves the memory cost of these variables. See the [output block](https://epochpic.github.io/documentation/input_deck/input_deck_output_block.html) documentation
- Do you need _every_ particle? You can force the code to only output a subset of the particles present, using the [subset block](https://epochpic.github.io/documentation/input_deck/input_deck_subset.html). With persistent subsets, you can make sure the same particles are present each dump for particle tracking.

Hope this helps,
Stuart
